src/mds/MDCache.cc

```diff
@@ -9940,6 +9940,21 @@ void MDCache::request_cleanup(const MDRequestRef& mdr)
 
   mdr->dead = true;
 
+  if (mdr->killed && mdr->client_request && mdr->is_batch_head()) {
+    dout(10) << "request " << *mdr << " was killed and dead" << dendl;
+    //if the mdr is a "batch_op" and it has followers, pick a follower as
+    //the new "head of the batch ops" and go on processing the new one.
+    int mask = mdr->client_request->head.args.getattr.mask;
+    auto it = mdr->batch_op_map->find(mask);
+    auto new_batch_head = it->second->find_new_head();
+    if (!new_batch_head) {
+      mdr->batch_op_map->erase(it);
+      return;
```

Comment:
I don't think we should return in this flow, should we?

Reply:
Yeah, correct. I will fix it. The test is still running locally.